Modify: GitHub.cli version 2.63.0

aisgbnok opened the pull request:
Checklist for Pull Requests
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/194873)

mdanish-kh commented on manifests/g/GitHub/cli/2.63.0/GitHub.cli.installer.yaml:
```diff
@@ -14,22 +14,22 @@ InstallModes:
   ProductCode: '{FAAF4306-25F7-422D-9758-CDFD6E6FC729}'
   AppsAndFeaturesEntries:
-  - ProductCode: '{C08A19D6-620B-415D-B4FA-0224B3D6F78E}'
+  - ProductCode: '{FAAF4306-25F7-422D-9758-CDFD6E6FC729}'
     UpgradeCode: '{8CFB9531-B959-4E1B-AA2E-4AF0FFCC4AF4}'
```
```suggestion
  - UpgradeCode: '{8CFB9531-B959-4E1B-AA2E-4AF0FFCC4AF4}'
```

aisgbnok replied:
I see, so there is no functional reason to include `ProductCode` twice. As long as it is at the root node it is fine? [I include it twice in manifests because the documentation states it should be when ARP node is present.](https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.6.0/installer.md#:~:text=the%20ProductCode%20should%20be%20placed%20both%20within%20the%20installer%20and%20the%20AppsAndFeaturesEntries) I have wondered if it was necessary before.

mdanish-kh replied:
I believe that recommendation came from a time (probably WinGet 1.3) when CLI wouldn't match against the root value properly so the community decided to have it both at ARP and Installer level. But that should be not be the case anymore. In general, avoid ARP fields when there isn't a whole lot need for them (i.e., Package is having matching issues)

aisgbnok replied:
Ahh I see. Thank you very much, @mdanish-kh. I will update this PR sometime today, and take note of your feedback for the future as well.

mdanish-kh commented on manifests/g/GitHub/cli/2.63.0/GitHub.cli.installer.yaml:
```diff
@@ -14,22 +14,22 @@ InstallModes:
   ProductCode: '{52F2E2D9-E7E7-48AE-B12A-875739D6BB61}'
   AppsAndFeaturesEntries:
-  - ProductCode: '{8937394A-B572-4DB3-9ECC-CD82153E3754}'
+  - ProductCode: '{52F2E2D9-E7E7-48AE-B12A-875739D6BB61}'
     UpgradeCode: '{767EC5D2-C8F0-4912-9901-45E21F59A284}'
```
```suggestion
  - UpgradeCode: '{767EC5D2-C8F0-4912-9901-45E21F59A284}'
```

Remove ProductCode from ARP level entirely. Some manifest creator tools do not update this automatically leaving old values getting carried forward in manifests and then having to make manual PRs to fix. It's defined with correct values in the Installer level which suffices.

ItzLevvie commented:
[Policy] [Changes-Requested](https://github.com/microsoft/winget-pkgs/labels/Changes-Requested)

ItzLevvie commented:
[Policy] Changes-Requested

ItzLevvie commented:
[Policy] Needs-Author-Feedback

ItzLevvie reviewed (changes requested):
See https://github.com/microsoft/winget-pkgs/pull/194873#pullrequestreview-2470017196

ItzLevvie commented:
[Policy] Needs-Author-Feedback

aisgbnok committed:
Modify: GitHub.cli version 2.63.0

aisgbnok committed:
Remove redundant ProductCode from ARP node

The installer-level definition suffices; defining it again in the ARP node is unnecessary and can cause inconsistencies.

Co-authored-by: Muhammad Danish <mdanishkhdev@gmail.com>
